cpp/kiss_icp/core/Preprocessing.hpp

```diff
@@ -28,6 +28,9 @@
 #include <vector>
 
 namespace kiss_icp {
+/// Voxelize point cloud keeping the original coordinates
```

Comment:
One question to discuss now in this particular PR:

Assuming we get rid of `CorrectKITTIScan`, then this "Preprocessing" module will only have two functions:

```cpp
std::vector<Eigen::Vector3d> VoxelDownsample(const std::vector<Eigen::Vector3d> &frame,
                                             const double voxel_size);

std::vector<Eigen::Vector3d> Preprocess(const std::vector<Eigen::Vector3d> &frame,
                                        double max_range,
                                        double min_range);
```

The `Preprocess` function, which is only cropping (so we could change the name of it), is only consumed in the kiss-icp pipeline, so potentially, we could leave it there.

This makes me think `VoxelDownsample` is the only missing function here, and we can put it in our new and shiny `VoxelUtiils.hpp` module.

Reply:
Moving this to a separate PR now https://github.com/PRBonn/kiss-icp/pull/369